File: gateOne/Python codes/ATM_app.py
import random

def ATM_app():
    account_numbers = []
    balances = []
    pins = []

    print("Welcome")
    print("Kindly type Next to select an option:")
    next_option = input()

    transactions = True

    while transactions:
        print("\n--- ATM Machine ---")
        print("1. Create an account")
        print("2. Close account")
        print("3. Deposit money")
        print("4. Withdraw money")
        print("5. Check Account balance")
        print("6. Transfer from one account to another")
        print("7. Change Pin")
        print("8. Exit")

        choice = int(input("Choose an option: "))

        if choice == 1:
            print("Create account")
            first_name = input("Enter first name: ")
            last_name = input("Enter last name: ")
            pin = input("Enter a 4-digit PIN: ")

            if len(pin) != 4:
                print("Invalid PIN. Account creation failed.")
                continue

            account_number = random.randint(100000, 999999)
            account_numbers.append(account_number)
            balances.append(0.0)
            pins.append(pin)
            print("Account successfully created")
            print(f"Your account number is: {account_number}")

        elif choice == 2:
            account_number = int(input("Enter your account number: "))
            if account_number in account_numbers:
                index = account_numbers.index(account_number)
                print(f"Account number {account_number} closed successfully.")
                account_numbers.pop(index)
                balances.pop(index)
                pins.pop(index)
            else:
                print("Account not found.")

        elif choice == 3:
            account_number = int(input("Enter your account number: "))
            if account_number in account_numbers:
                index = account_numbers.index(account_number)
                amount = float(input("Enter deposit amount: "))
                if amount > 0:
                    balances[index] += amount
                    print(f"Deposit successful! New balance: ${balances[index]}")
                else:
                    print("Invalid deposit amount.")
            else:
                print("Account not found.")

        elif choice == 4:
            account_number = int(input("Enter your account number: "))
            if account_number in account_numbers:
                index = account_numbers.index(account_number)
                amount = float(input("Enter withdrawal amount: "))
                if 0 < amount <= balances[index]:
                    balances[index] -= amount
                    print(f"Withdrawal successful! New balance: ${balances[index]}")
                else:
                    print("Invalid withdrawal amount or insufficient funds.")
            else:
                print("Account not found.")

        elif choice == 5:
            account_number = int(input("Enter your account number: "))
            if account_number in account_numbers:
                index = account_numbers.index(account_number)
                print(f"Your current balance is: ${balances[index]}")
            else:
                print("Account not found.")

        elif choice == 6:
            account_number = int(input("Enter your account number: "))
            if account_number in account_numbers:
                index = account_numbers.index(account_number)
                recipient_account = int(input("Enter the recipient's account number: "))
                if recipient_account in account_numbers:
                    recipient_index = account_numbers.index(recipient_account)
                    transfer_amount = float(input("Enter transfer amount: "))

                    if 0 < transfer_amount <= balances[index]:
                        balances[index] -= transfer_amount
                        balances[recipient_index] += transfer_amount
                        print("Transfer successful!")
                        print(f"Your new balance: ${balances[index]}")
                        print(f"Recipient's new balance: ${balances[recipient_index]}")

                        generate_receipt = input("Would you like to generate a receipt? (yes/no): ")
                        if generate_receipt.lower() == "yes":
                            print("\n---------- Transfer Receipt --------------------------")
                            print(f"Sender Account: {account_number}")
                            print(f"Recipient Account: {recipient_account}")
                            print(f"Transfer Amount: ${transfer_amount}")
                            print("------------------------------------------------------------\n")
                    else:
                        print("Invalid transfer amount or insufficient funds.")
                else:
                    print("Recipient account not found.")
            else:
                print("Your account number not found.")


ATM_app()

File: gateOne/gtbBank/ATMApp.java
import java.util.ArrayList; 
import java.util.Random;
import java.util.Scanner;

public class AtmApp {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Random random = new Random();
        ArrayList<Integer> accountNumbers = new ArrayList<>();
        ArrayList<Double> balances = new ArrayList<>();
        ArrayList<String> pins = new ArrayList<>();

	System.out.println("Welcome");
	System.out.println("Kindly type 'Next' to select an option:");
	String next = scanner.next();
	if (!next.equals("Next")) {
	System .out.println("Please type 'Next' only");
	next = scanner.next();
					
	}
        boolean transactions = true;

        while (transactions) {
            System.out.println("\n--- ATM Machine ---");
            System.out.println("1. Create an account");
            System.out.println("2. Close account");
            System.out.println("3. Deposit money");
            System.out.println("4. Withdraw money");
            System.out.println("5. Check Account balance");
            System.out.println("6. Transfer from one account to another");
            System.out.println("7. Change Pin");
            System.out.println("8. Exit");

            System.out.print("Choose an option: ");
            int choice1 = scanner.nextInt();

            switch (choice1) {

                case 1:
                    System.out.println("Create account ");
                    System.out.println("Enter first name: ");
                    String firstName = scanner.next();

                    System.out.println("Enter last name: ");
                    String lastName = scanner.next();
                    System.out.print("Enter a 4-digit PIN: ");
                    String pin = scanner.next();

                    if (pin.length() != 4) {
                        System.out.println("Invalid PIN. Account creation failed.");
                        break;
                    }

                    int accountNumber = random.nextInt(900000);
                    accountNumbers.add(accountNumber);
                    balances.add(0.0);
                    pins.add(pin);
                    System.out.println("Account successfully created");
                    System.out.println("Your account number is: " + accountNumber);
                     break;

                case 2:
                    System.out.print("Enter your account number: ");
                    accountNumber = scanner.nextInt();
                    int index = accountNumbers.indexOf(accountNumber);
                    if (index != -1) {
                        System.out.printf("Account number %d closed successfully.\n", accountNumber);
                        accountNumbers.remove(index);
                        balances.remove(index);
                        pins.remove(index);
                    } else {
                         System.out.println("Account not found.");
                    }
                    break;

                case 3:
                     System.out.print("Enter your account number: ");
                    accountNumber = scanner.nextInt();
                    index = accountNumbers.indexOf(accountNumber);
                    if (index != -1) {
                        System.out.print("Enter deposit amount: ");
                        double amount = scanner.nextDouble();
                        if (amount > 0) {
                            balances.set(index, balances.get(index) + amount);
                            System.out.println("Deposit successful! New balance: $" + balances.get(index));
                        } else {
                            System.out.println("Invalid deposit amount.");
                        }
                    } else {
                        System.out.println("Account not found.");
                    }
                    break;

                case 4:
                    System.out.print("Enter your account number: ");
                    accountNumber = scanner.nextInt();
                    index = accountNumbers.indexOf(accountNumber);
                    if (index != -1) {
                        System.out.print("Enter withdrawal amount: ");
                        double amount = scanner.nextDouble();
                        if (amount > 0 && amount <= balances.get(index)) {
                            balances.set(index, balances.get(index) - amount);
                            System.out.println("Withdrawal successful! New balance: $" + balances.get(index));
                        } else {
                            System.out.println("Invalid withdrawal amount or insufficient funds.");
                        }
                    } else {
                        System.out.println("Account not found.");
                    }
                    break;

                case 5:
                    System.out.print("Enter your account number: ");
                    accountNumber = scanner.nextInt();
                    index = accountNumbers.indexOf(accountNumber);
                    if (index != -1) {
                        System.out.println("Your current balance is: $" + balances.get(index));
                    } else {
                        System.out.println("Account not found.");
                    }
                    break;

               case 6:
			
   			 System.out.print("Enter your account number: ");
   			 accountNumber = scanner.nextInt();
    			index = accountNumbers.indexOf(accountNumber);
    			System.out.println("Create recipient's account first if you have not:");
   			 if (index != -1) {
       				 System.out.println ("Enter the recipient's account number: ");
       				 int recipientAccount = scanner.nextInt();
       				 int recipientIndex = accountNumbers.indexOf(recipientAccount);

      			  if (recipientIndex != -1) {
            			System.out.print("Enter transfer amount: ");
           			 double transferAmount = scanner.nextDouble();

            		if (transferAmount > 0 && transferAmount <= balances.get(index)) {
                		balances.set(index, balances.get(index) - transferAmount);
                		balances.set(recipientIndex, balances.get(recipientIndex) + transferAmount);
                
               		 System.out.println("Transfer successful!");
               		 System.out.println("Your new balance: $" + balances.get(index));
                	System.out.println("Recipient's new balance: $" + balances.get(recipientIndex));

               		 System.out.print("Would you like to generate a receipt? (yes/no): ");
               		 String generateReceipt = scanner.next();
				
			if (generateReceipt.equalsIgnoreCase("yes")) {
                    	System.out.println("\n---------- Transfer Receipt --------------------------");
                    	System.out.println("Sender Account: " + accountNumbers.get(index));
                   	 System.out.println("Recipient Account: " + accountNumbers.get(recipientIndex));
                    	System.out.println("Transfer Amount: $" + transferAmount);
                   	 System.out.println("------------------------------------------------------------\n");
               			 }
           		 } else {
               		 System.out.println("Invalid transfer amount  or insufficient funds.");
           		 }
       			 } else {
           		 System.out.println("Recipient account not found.");
       		      		 }


               	 
   			 } else {
       			 System.out.println("Your account number not found.");
    				}
   			 break;

		case 7:
                    System.out.print("Enter your account number: ");
                    accountNumber = scanner.nextInt();
                    index = accountNumbers.indexOf(accountNumber);
                    if (index != -1) {
                        System.out.print("Enter your current PIN: ");
                        String currentPin = scanner.next();
                        if (pins.get(index).equals(currentPin)) {
                            System.out.print("Enter your new 4-digit PIN: ");
                            String newPin = scanner.next();
                            System.out.print("Confirm your new 4-digit PIN: ");
                            String confirmPin = scanner.next();
                            if (confirmPin.length() == 4 && newPin.equals(confirmPin)) {
                                pins.set(index, confirmPin);
                                System.out.println("PIN changed successfully.");
                            } else {
                                System.out.println("PINs do not match or invalid PIN. PIN change failed.");
                            }
                        } else {
                            System.out.println("Incorrect current PIN.");
                        }
                    } else {
                        System.out.println("Account not found.");
                    }
                    break;

                case 8:
                    transactions = false;
                    System.out.println("Thank you for using the ATM. Goodbye!");
                    break;

                default:
                    System.out.println("Invalid option. Please try again.");
                    break;
            }
        }
    }
}
